### PR TITLE
refactor: Add cache for prism plugin

### PR DESCRIPTION
**Pull request by Saul-Mirone**
Add cache to prevent recalculate code blocks not modified.

### CODE REVIEW

File: src/plugins/Prism.ts [hunk 1]
```diff
@@ -60,27 +63,42 @@ function getDecorations({ doc, name }) {
...
-        return {
-          ...node,
-          from,
-          to,
-        };
-      })
-      .forEach(node => {
-        const decoration = Decoration.inline(node.from, node.to, {
-          class: (node.classes || []).join(" "),
+    if (!cache[block.pos] || !cache[block.pos].node.eq(block.node)) {
```
Comment (tommoor) [inline src/plugins/Prism.ts:66]: We should move `refractor.highlight` inside the `if` statement, right?!

**Comment by tommoor** [comment]
Thanks for the contribution here

File: src/plugins/Prism.ts (hunk 1 above)
Comment (Saul-Mirone) [reply src/plugins/Prism.ts:64]: Fixed

**Review by tommoor — APPROVED** [review]

**Comment by Saul-Mirone** [comment]
Cool, can't wait to see this PR been merged.

**Comment by tommoor** [comment]
All good in my testing – are you using the editor for a project?

**Comment by Saul-Mirone** [comment]
> All good in my testing – are you using the editor for a project?

No, I just want to write my own wysiwyg markdown editor using prosemirror and typescript, for learning purpose. This project gives me many inspirations.

**Comment by Saul-Mirone** [comment]
BTW, here is my project [milkdown](https://github.com/Saul-Mirone/milkdown), it's still a tiny one.

**Pull request merged**